submissions/zhenyakornilov/memory-pair-game/js/index.js

```diff
@@ -0,0 +1,107 @@
+import cards from "./cards.js";
+
+const cardsMenu = document.getElementById("game-cards");
+const totalTriesCounter = document.getElementById("total-flips");
+const states = {
+  cardsMatches: 0,
+  firstFlippedCard: null,
+  secondFlippedCard: null,
+  currentCardPair: [],
+  totalTries: 0,
+};
+const allGameCards = shuffleCards([...cards, ...cards]);
+renderCards(allGameCards, cardsMenu);
+
+function shuffleCards(cardsArr) {
+  for (let i = cardsArr.length - 1; i > 0; i--) {
+    let j = Math.floor(Math.random() * (i + 1));
+    let swap = cardsArr[i];
+    cardsArr[i] = cardsArr[j];
+    cardsArr[j] = swap;
+  }
+
+  return cardsArr;
+}
+
+function renderCards(cardsObj, cardsList) {
+  let cardsHTML = cardsObj
+    .map(({ name, imageSrc }) => {
+      return `
+      <li class="game-card">
+        <div class="card-inner">
+          <div class="card-front">
+            <img class="card-img-front" src="images/treble_clef.webp" alt="image of treble clef">
+          </div>
+          <div class="card-back">
+            <img class="card-img-back" src="${imageSrc}" alt="image of ${name}">
+            <p class="card-desc">${name}</p>
+          </div>
+        </div>
+      </li>
+      `;
+    })
+    .join("");
+
+  cardsList.innerHTML = cardsHTML;
+}
+
+function sleep(milliseconds) {
+  return new Promise((resolve) => setTimeout(resolve, milliseconds));
+}
+
+function flipBack(currentCardPair) {
+  currentCardPair.forEach((card) => {
+    if (!card.classList.contains("matched")) {
+      card.closest(".game-card").classList.remove("flipped");
+    }
+  });
+}
+
+function checkCardsMatch() {
+  return states.firstFlippedCard.isEqualNode(states.secondFlippedCard);
```

Comment:
why so a complicated approach? I guess better compare name of the cards, even for performance reasons